features/daily-image.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const axios = require('axios');
+
+module.exports = function(controller) {
+ 
+  controller.on('slash_command', async (bot, message) => {
+    console.log(message);
+    if (message.command == '/daily-image') {
+      let imageSearchString = getImageQueryString(message.text.trim());
+      console.log('Image Search String = '+imageSearchString);
+      //This is an interesting workaround to handle timing of the nested promises
+      await bot.changeContext(message.reference);
```

Comment:
Is changeContext() async? Is that related to the workaround being described?

Reply:
I believe the bot is now getting a response within its request timeout to set its original context so any chained "slow" responses won't be revoked.